### ISSUE TITLE
Rule violation in valid use case (role="gridcell" aria-roledescription="button")

### ISSUE THREAD

**Issue by williamernest** [dequelabs/axe-core#3381]
Product: axe-core@4.4

Expectation: No error

Actual:
```
    Violated rule: "aria-roledescription"
    Reasoning: "aria-roledescription must be on elements with a semantic role"
```
Motivation: Tested on various browsers/screen readers and behavior is expected (correctly reads roledescription)
VO/Chrome : Correctly reads roledescription
VO/Firefox: Correctly reads roledescription
VO/Safari: Correctly reads roledescription
NVDA/Chrome: Correctly reads roledescription
NVDA/Firefox: Correctly reads roledescription

https://codepen.io/williamernest/full/GROjddd

**Comment by WilcoFiers** [comment]
Are you seeing this reported as a violation? I am seeing this come up as "needs review". If you're using the axe extension, you can tell from the flag icon.

The reason axe-core flags this is because this will not work in internet explorer with JAWS. Depending on how it's used, missing information from aria-roledescription can cause significant problems. Use of IE is a good bit higher in among screen reader users than it is in the general population, so it's something just to keep an eye out for.

**Comment by WilcoFiers** [comment]
No response. We have done the testing, there are definitely still cases where aria-roledescription in combination with role=gridcell causes issues. Going to leave this as is.
